common/common.go

```diff
@@ -136,66 +135,47 @@ func NewHTTPClientWithTimeout(t time.Duration) *http.Client {
 	return h
 }
 
-// StringSliceDifference concatenates slices together based on its index and
-// returns an individual string array
-func StringSliceDifference(slice1, slice2 []string) []string {
-	var diff []string
-	for i := range 2 {
-		for _, s1 := range slice1 {
-			found := false
-			for _, s2 := range slice2 {
-				if s1 == s2 {
-					found = true
-					break
-				}
-			}
-			if !found {
-				diff = append(diff, s1)
-			}
+// SliceDifference returns the elements that are in slice1 or slice2 but not in both
+func SliceDifference[T comparable](slice1, slice2 []T) []T {
+	diff := make([]T, 0, len(slice1)+len(slice2))
```

Comment:
I gave you a silly billy suggestion. It gets gets weird when there are duplications, and its modifying the incoming slice. 
Just the prealloc is fine. 🙏 
```diff
diff --git a/common/common.go b/common/common.go
index d0799b199..57e0a92d2 100644
--- a/common/common.go
+++ b/common/common.go
@@ -138,16 +138,12 @@ func NewHTTPClientWithTimeout(t time.Duration) *http.Client {
 // SliceDifference returns the elements that are in slice1 or slice2 but not in both
 func SliceDifference[T comparable](slice1, slice2 []T) []T {
        diff := make([]T, 0, len(slice1)+len(slice2))
-       target := 0
        for x := range slice1 {
                if !slices.Contains(slice2, slice1[x]) {
                        diff = append(diff, slice1[x])
                        continue
                }
-               slice1[target] = slice1[x]
-               target++
        }
-       slice1 = slice1[:target]
        for x := range slice2 {
                if !slices.Contains(slice1, slice2[x]) {
                        diff = append(diff, slice2[x])
```